editor/sidebar/table-of-content/index.js

```diff
@@ -0,0 +1,52 @@
+/**
+ * External dependencies
+ */
+import { connect } from 'react-redux';
+import { filter } from 'lodash';
+import classnames from 'classnames';
+
+/**
+ * WordPress Dependencies
+ */
+import { __, sprintf } from 'i18n';
+import { PanelBody } from 'components';
+
+/**
+ * Internal Dependencies
+ */
+import './style.scss';
+import { getBlocks } from '../../selectors';
+
+const TableOfContents = ( { blocks } ) => {
+	const headings = filter( blocks, ( block ) => block.name === 'core/heading' );
```

Comment:
Minor: `filter` has nice convenience for property matching:

```js
const headings = filter( blocks, { name: 'core/heading' } );
```

Though supposedly slated to be removed by default in next major: https://github.com/lodash/lodash/wiki/Roadmap#v500-2017